### ISSUE TITLE
[request] pegparser/2.1.1

### ISSUE THREAD

**Issue by dfontenot** [conan-io/conan-center-index#4476]
### Package Details
  * Package Name/Version: **pegparser/2.1.1**
  * Source code: **https://github.com/TheLartians/PEGParser**

### Description Of The Library / Tool

C++17 parser generator. I'm willing to add it myself (would be my first contribution)--looking to help out.

**Comment by Croydon** [comment]
> I'm willing to add it myself (would be my first contribution)--looking to help out.

Pull requests are always welcome. Feel free to go ahead

**Comment by SpaceIm** [comment]
You'll have to remove CPM mechanism from upstream CMakeLists.

**Comment by dfontenot** [comment]
@SpaceIm good call out, I did not see that dependency explicitly called out. I can start with that one. In general, should there always be a package request issue created for each package to add to the CCI? When I made this issue, that was my understanding of the contributors' guide.

**Comment by SpaceIm** [comment]
AFAIK, it's not required. You can open a PR without an associated issue.

**Comment by TheLartians** [comment]
Hey, I just discovered this issue by accident, happy to see my library added to Conan!

Unfortunately, I have no experience with Conan myself, but one of the goals of CPM.cmake is to be easy to integrate with external PMs. E.g. if you set the CMake or Env variable `CPM_USE_LOCAL_PACKAGES`, then `CPM.cmake` will try `find_package()` before downloading the dependency. If you need another way to override CPM.cmake behaviour for easier Conan integration please let me know!

**Comment by SpaceIm** [comment]
@TheLartians Thanks for the tips.

Other problem: in https://github.com/TheLartians/PEGParser/blob/master/cmake/CPM.cmake, at configure time it downloads CPM.cmake file from another github repo unless found locally. It should be avoided in conan-center.
I guess we could first create a package for CPM 0.28.0 (build requirement of PEGParser 2.2.0, to avoid to patch pegparser's CMakeLists).
We can't remove `include(cmake/CPM.cmake)`, otherwise CMake CPMAddPackage 

Then we could set `CPM_SOURCE_CACHE` to point to a parent directory of CPM.cmake.

Oh, and I also see a PackageProject.cmake project downloaded for installation, we also have to create a recipe for this one.

**Comment by TheLartians** [comment]
Good catch! Yeah, creating a recipe for CPM.cmake and setting `CPM_SOURCE_CACHE` sounds like a good approach to for projects using CPM.cmake dependencies to be added to Conan. That way no source modification should be necessary.

On that topic, as CPM.cmake uses GitHub actions for release management, would it make sense to use [Minimonium/conan-request-action](https://github.com/Minimonium/conan-request-action) for each release to notify folks of an update? Or is there even a way to update the Conan recipe directly from GitHub actions?
